web/packages/extension/src/popup.ts

```diff
@@ -187,5 +186,30 @@ window.addEventListener("DOMContentLoaded", () => {
         window.close();
     });
 
-    displayTabStatus();
+    await displayTabStatus();
+    const permissionsButton = document.getElementById(
+        "permissions-button",
+    ) as HTMLButtonElement;
+    permissionsButton.textContent = utils.i18n.getMessage(
+        "grant_single_site_permission",
+    );
+    const url = activeTab?.url ? new URL(activeTab.url) : null;
+    if (
+        url &&
+        ["https:", "http:"].includes(url.protocol) &&
+        !(await utils.hasHostPermissionForActiveTab())
+    ) {
+        permissionsButton.classList.remove("hidden");
+        permissionsButton.addEventListener("click", async () => {
+            const grant = await utils.permissions.request({
```

Comment:
I think there is actually a way of doing this reliably by sending a message to the background script, which requests the permission and reloads the tab on success.

(I only tested this in Firefox)

Reply:
I thought that Firefox would forward the user activation via sendMessage, but it seems like it probably doesn't. Still not sure why I thought this worked in my testing ...

Reply:
Discussion on Discord leads me to believe there's no easy way to do this in Firefox right now. If anyone thinks of something that can be a follow up PR.

Reply:
See https://github.com/ruffle-rs/ruffle/pull/16808/commits/9152551e245509d4d33494c4e6754986ca2997e8